pkg/apis/apps/validation/validation_test.go

```diff
@@ -42,6 +42,96 @@ func intStrAddr(intOrStr intstr.IntOrString) *intstr.IntOrString {
 	return &intOrStr
 }
 
+type setTweak func(s apps.StatefulSet)
+
+func makeSet(selectorLabels *map[string]string, template *api.PodTemplate, tweaks ...setTweak) apps.StatefulSet {
```

Comment:
I don't think you need a pointer to the map - just passing the map "by value" is sufficient.

But also, you could make a tweak func?  Set a sane default in the `makeStatefulSet` func, and call the tweak ONLY from the cases that need a specific selector (which emphasizes that the selector is important) like:

```
fun setSelector(labels map[string]string) tweakStatefulSet {
    return func(ss apps.StatefulSet) {
        ss.Spec.Selector = &metav1.LabelSelector{MatchLabels: &labels)}
    }
}
```

Reply:
modified as suggested